StringOperations.java

```diff
@@ -136,11 +120,108 @@ public static void main(String[] args) {
             } else {
                 System.out.print("Would you like to try again? (y/n): ");
             }
-        } while (scanner.nextLine().equalsIgnoreCase("yes"));
+        } while (scanner.nextLine().equalsIgnoreCase("y"));
         System.out.println("Program terminated.");
         scanner.close();
     }
+
+    // Split
+    public static String[] split(String inputString, String pattern) {
+        return inputString.split(pattern);
+    }
+
+    // Max Repeat
+    public static char maxRepeat(String inputString) {
+        int[] freq = new int[256];
+        for (int i = 0; i < inputString.length(); i++) {
+            freq[inputString.charAt(i)]++;
+        }
+        int max = -1;
+        char result = ' ';
+        for (int i = 0; i < inputString.length(); i++) {
+            if (freq[inputString.charAt(i)] > max) {
+                max = freq[inputString.charAt(i)];
+                result = inputString.charAt(i);
+            }
+        }
+        return result;
+    }
+
+    // Sort
+    public static String sort(String inputString) {
+        char[] chars = inputString.toCharArray();
+        Arrays.sort(chars);
+        return new String(chars);
+    }
+
+    // Shift
+    public static String shift(String inputString, int n) {
+        if (n > inputString.length()) {
```

Comment:
The condition if (n > inputString.length()) should be replaced with if (n >= inputString.length()) because shifting by the exact length of the string results in the same string.